detector/vuls2/vuls2.go

```diff
@@ -0,0 +1,633 @@
+)
+
+func Detect(r *models.ScanResult, vuls2Cnf config.Vuls2DictConf, noProgress bool) error {
+	switch r.Family {
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+	case constant.Oracle:
+		// FIXME: for testing, remove later
+	default:
+		return nil
+	}
+
+	c := Config{
+		Repository: func() string {
+			if vuls2Cnf.Repository != "" {
+				return vuls2Cnf.Repository
+			}
+			return DefaultGHCRRepository
+		}(),
+		Path: func() string {
+			if vuls2Cnf.Path != "" {
+				return vuls2Cnf.Path
+			}
+			return DefaultPath
+		}(),
+		SkipUpdate: vuls2Cnf.SkipUpdate,
+		Quiet:      noProgress,
+	}
+
+	if err := c.Refresh(); err != nil {
+		return xerrors.Errorf("Refresh vuls2 db: %w", err)
+	}
+
+	dbc, err := c.New()
+	if err != nil {
+		return xerrors.Errorf("Failed to get new db connection. err: %w", err)
+	}
+	if err := dbc.Open(); err != nil {
+		return xerrors.Errorf("Failed to open db. err: %w", err)
+	}
+	defer dbc.Close()
+
+	meta, err := dbc.GetMetadata()
+	if err != nil || meta == nil {
+		return xerrors.Errorf("Failed to get metadata. err: %w", err)
+	}
+	if meta.SchemaVersion < db.SchemaVersion {
+		return xerrors.Errorf("schema version is old. expected: %q, actual: %q", db.SchemaVersion, meta.SchemaVersion)
+	}
+
+	newInput := preConvert(r)
+
+	dres, err := detect(dbc, newInput)
+	if err != nil {
+		return xerrors.Errorf("Failed to detect. err: %w", err)
+	}
+
+	e, err := ecosystemTypes.GetEcosystem(r.Family, r.Release)
+	if err != nil {
+		return xerrors.Errorf("Failed to get ecosystem. family: %s, release: %s, err: %w", r.Family, r.Release, err)
+	}
+
+	vulnInfos := postConvert(e, dres, r)
+	r.ScannedCves = vulnInfos
+	return nil
+}
+
+func preConvert(old *models.ScanResult) scanTypes.ScanResult {
+	pkgs := map[string]scanTypes.OSPackage{}
+	for _, p := range old.SrcPackages {
+		for _, bn := range p.BinaryNames {
+			pkgs[bn] = scanTypes.OSPackage{
+				SrcName:    p.Name,
+				SrcVersion: p.Version,
+			}
+		}
+	}
+	for _, p := range old.Packages {
+		base := pkgs[p.Name]
+		base.Name = p.Name
+		base.Version = p.Version
+		base.Release = p.Release
+		base.NewVersion = p.NewVersion
+		base.NewRelease = p.NewRelease
+		base.Arch = p.Arch
+		base.Repository = p.Repository
+		base.ModularityLabel = p.ModularityLabel
+		pkgs[p.Name] = base
+	}
+
+	return scanTypes.ScanResult{
+		JSONVersion: 0,
+		ServerName:  old.ServerName,
+		Family:      old.Family,
+		Release:     old.Release,
+
+		Kernel: scanTypes.Kernel{
+			Release:        old.RunningKernel.Release,
+			Version:        old.RunningKernel.Version,
+			RebootRequired: old.RunningKernel.RebootRequired,
+		},
+		OSPackages: func() []scanTypes.OSPackage {
+			ps := make([]scanTypes.OSPackage, 0, len(pkgs))
+			for _, p := range pkgs {
+				ps = append(ps, p)
+			}
+			return ps
+		}(),
+		CPE: func() []string {
+			s, ok := old.Config.Scan.Servers[old.ServerName]
+			if !ok {
+				return nil
+			}
+			return s.CpeNames
+		}(),
```

Comment:
CPE detection is not performed on the nightly side, so it does not seem necessary.
```suggestion
```